apollo-router/src/layers/async_checkpoint.rs

```diff
@@ -277,21 +156,21 @@ mod async_checkpoint_tests {
         let expected_label = "from_mock_service";
 
         let mut execution_service = MockExecutionService::new();
-        execution_service.expect_clone().return_once(move || {
-            let mut execution_service = MockExecutionService::new();
-            execution_service
-                .expect_call()
-                .times(1)
-                .returning(move |req: ExecutionRequest| {
-                    Ok(ExecutionResponse::fake_builder()
-                        .label(expected_label.to_string())
-                        .context(req.context)
-                        .build()
-                        .unwrap())
-                });
-
-            execution_service
-        });
+
+        execution_service
+            .expect_clone()
+            .return_once(MockExecutionService::new);
+
+        execution_service
+            .expect_call()
+            .times(1)
+            .returning(move |req| {
+                Ok(ExecutionResponse::fake_builder()
+                    .label(expected_label.to_string())
+                    .context(req.context)
+                    .build()
+                    .unwrap())
+            });
 
         let service_stack = ServiceBuilder::new()
```

Comment:
Let's create a ticket to migrate custom layers to use tower_test.

Reply:
https://apollographql.atlassian.net/browse/ROUTER-984